src/internal/observable/dom/fetch.ts

```diff
@@ -51,8 +94,14 @@ import { Subscription } from '../../Subscription';
  * @returns An Observable, that when subscribed to performs an HTTP request using the native `fetch`
  * function. The {@link Subscription} is tied to an `AbortController` for the the fetch.
  */
-export function fromFetch(input: string | Request, init?: RequestInit): Observable<Response> {
-  return new Observable<Response>(subscriber => {
+export function fromFetch<T>(
+  input: string | Request,
+  initWithSelector: RequestInit & {
+    selector?: (response: Response) => ObservableInput<T>
+  } = {}
+): Observable<Response | T> {
+  const { selector, ...init } = initWithSelector;
```

Comment:
A observation. Now that `init` is created using spread, this means that `init` always will be an object, even it has no fields. And below wrap with `if(init)` has no value

https://github.com/ReactiveX/rxjs/pull/5306/files#diff-717ea4ff7feb7c6009e9048143f3ef0fc9ddc6cdeb16b1effcb19d18f3a9c875R119

Reply:
@iamandrewluca 👍 Yep. Feel free to open a PR to simplify it.